File: to_sort/utils/getLigands.pl
#!/usr/bin/perl
use warnings;

$file=$ARGV[0];


open(IN,"$file");
$i=0;$n=0;$check=0;
$prevprev="";
$prev ="";

while(<IN>){
	chomp;
	$line=$_;
	
	if($prevprev =~ /i_i_glide_posenum/ && $prev =~ /\:\:\:/){
		$lig[$i] = $line;
		$i++;
	}
	if ($prevprev =~ /i_i_glide_posenum/ && $prev !~ /\:\:\:/) {$check=1;}
	if ($prev =~ /\:\:\:/ && $check==1){
		$lig[$i] = $line;
		$check=0;
		$i++;
	}
	if($n > 2){$prevprev=$prev};
	if($n > 1){$prev=$line};
	$n++;
	
}
close(IN);

print (join("\n",@lig),"\n");

File: to_sort/utils/getLigands.py
import sys
import gzip

file_name = sys.argv[1]

f=gzip.open(file_name)
all_lines =f.readlines()
f.close()

ligands = []

for i,line in enumerate(all_lines):
	if 'i_i_glide_posenum' in line:
		ligands.append(int(all_lines[i+2].strip()))

for lig in ligands:
	print lig	